packages/devextreme/js/__internal/grids/grid_core/keyboard_navigation/m_keyboard_navigation.ts

```diff
@@ -688,14 +698,18 @@ export class KeyboardNavigationController extends modules.ViewController {
     }
   }
 
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  protected _toggleInertAttr(value = false): void {}
+
   protected _tabKeyHandler(eventArgs, isEditing) {
     const editingOptions = this.option('editing');
     const direction = eventArgs.shift ? 'previous' : 'next';
     const isCellPositionDefined = isDefined(this._focusedCellPosition)
       && !isEmptyObject(this._focusedCellPosition);
+    const isLastValidCell = !eventArgs.shift && this._isLastValidCell(this._focusedCellPosition);
 
     let isOriginalHandlerRequired = !isCellPositionDefined
-      || (!eventArgs.shift && this._isLastValidCell(this._focusedCellPosition))
+      || isLastValidCell
       || (eventArgs.shift && this._isFirstValidCell(this._focusedCellPosition));
```

Comment:
Minor refactoring [OPTIONAL].

Let's make the code more uniform and add the ```isFirstValidCell``` variable.
Something like:
```ts
const isFirstValidCell = eventArgs.shift && this._isFirstValidCell(this._focusedCellPosition);
const isLastValidCell = !eventArgs.shift && this._isLastValidCell(this._focusedCellPosition);

let isOriginalHandlerRequired = !isCellPositionDefined
      || isFirstValidCell
      || isLastValidCell;